### PR TITLE
start vignette

### PR DESCRIPTION
**Pull request by ronnyli**
Summary of changes:
- Added a vignette
- Updated the README
- Slightly modified `plot_partial_dependency()`

### CODE REVIEW

File: R/partial_dependency_aggregate.R [hunk 1]
```diff
@@ -199,11 +199,10 @@ run_partial_dependency <- function(feature_dt,
                                                ensemble_fcn = ensemble_fcn,
                                                ensemble_models = ensemble_models)
   if (plot) {
-    plot_list <- loop_plot_fcn(pd_list, plot_fcn)
```
Comment (ronnyli) [inline R/partial_dependency_aggregate.R]: facet_plot_fcn originally required plot_list as an input but this is no longer the case

File: R/partial_dependency_individual.R [hunk 2]
```diff
@@ -119,19 +119,15 @@ plot_partial_dependency <- function(pd) {
   feature_col <- pd[["feature"]][1]
   model_names <- unique(pd[["model"]])
   # Set size attributes for each model in graph
-  size_models <- rep(1, length(model_names))
+  size_models <- rep(0.5, length(model_names))
```
Comment (ronnyli) [inline R/partial_dependency_individual.R]: Reduce the size of other models to improve the ensemble model's visibility

File: R/partial_dependency_individual.R [hunk 3]
```diff
@@ -119,19 +119,15 @@ plot_partial_dependency <- function(pd) {
...
   names(size_models) <- model_names
   # Set type attributes for each model in graph
   type_models <- rep("dotdash", length(model_names))
   names(type_models) <- model_names
-  # Set color attributes for each model in graph
-  color_models <- palette(rainbow(length(model_names)))
```
Comment (ronnyli) [inline R/partial_dependency_individual.R]: for some reason, `rainbow` on my computer only occasionally returns a color palette. It's safer and also looks better if we remove the manual color palette.

File: R/partial_dependency_aggregate.R (hunk 1 above)
Comment (BenRollert) [inline R/partial_dependency_aggregate.R]: nice!

File: inst/doc/brightbox.Rmd [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+lm1$coefficients
+```
+```{r, fig.height = 3, fig.width = 7}
+# Note that the slopes of the plotted lines match the coefficients
+library(brightbox)
+pd <- run_partial_dependency(feature_dt = dt[, c("a", "b")],
+                             model_list = list(linear_model = lm1))
+```
+
+The advantage of partial dependency plots shine when there is a non-linear relationship between the feature in question and the response. In such cases, a linear model will hide the true relationship due to its arbitrary constraints. Ideally we would like to use a model with fewer constraints but maintain interpretability.
```
Comment (BenRollert) [inline inst/doc/brightbox.Rmd]: I would change the wording here: a linear model doesn't have arbitrary constraints if all assumptions hold. I would say, "due to a linear model's assumptions, it cannot directly model non-linear relationships." Also, would say "we would like to use a more flexible model with fewer assumptions but maintain interpretability"

File: R/partial_dependency_individual.R [hunk 5]
```diff
@@ -140,6 +136,5 @@ plot_partial_dependency <- function(pd) {
            ggplot2::geom_point() + ggplot2::geom_line() +
            ggplot2::scale_size_manual(values = size_models) +
            ggplot2::scale_linetype_manual(values = type_models) +
-           ggplot2::scale_colour_manual(values = color_models) +
            ggplot2::labs(x = feature_col, y = "Partial Dependence"))
```
Comment (BenRollert) [inline R/partial_dependency_individual.R]: might consider adding `theme_bw`. I find it's almost always preferred as default.

File: inst/doc/brightbox.Rmd [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+# Generate partial dependency plots for each feature
+pd <- run_partial_dependency(x, model_list = list(xgboost = xgb, nnet = nn),
+                             ensemble_colname = "ensemble", ensemble_fcn = median)
+```
+In the returned plot we can see how `medv` changes with respect to each feature for every model. As a result, there are quite a few things we can learn by inspecting the plot.
+
+1. We can easily see which features have a positive or negative relationship with the response and that the patterns are more complex than in the earlier example with linear regression (`rm` and `nox` for example).
+1. There are some features and some ranges within features where the two models are roughly in agreement. From this we can conclude that the plotted signal in those ranges is rather strong. (`age`, `rm`, `dis`)
+1. Some features have a wider range in the y-axis than others. This means that the feature had a larger impact on the response. In general, if a partial dependency plot is completely flat (zero slope) then the feature does not affect the model's predictions at all (an exception being when interaction effects are present).
+
```
Comment (BenRollert) [inline inst/doc/brightbox.Rmd]: might note that ICE plots are a solution this - and hint that they are coming :)

Comment (BenRollert) [inline inst/doc/brightbox.Rmd]: the exception is more when you have interaction effects that lead to contradiction - ie, when  response is increasing with respect to variable Xj when Xi = 0 and decreasing when Xi =1

**Pull request merged**